bot/bot.py

```diff
@@ -89,9 +92,32 @@ def clear(self) -> None:
         self._recreate()
         super().clear()
 
+    def remove_extensions(self) -> None:
+        """Remove all extensions and Cog to close bot. Copy from discord.py's own `close` for right closing order."""
+        for extension in tuple(self.extensions):
+            try:
+                self.unload_extension(extension)
+            except Exception:
+                pass
+
+        for cog in tuple(self.cogs):
+            try:
+                self.remove_cog(cog)
+            except Exception:
+                pass
```

Comment:
This is _almost_ an exact copy. I'm concerned that if you're copying their code, you must adhere to their license. This would mean including their license and copyright. Here's an example of how I did it for something unrelated in another PR https://github.com/python-discord/bot/blob/de592dc5eb22d061c9b988844e8c7d695a37fa58/LICENSE-THIRD-PARTY